In [9]:
import joblib

#Carregando o 'conhecimento' salvo
model = joblib.load('modelo_sentimento.pkl')
vetorizador = joblib.load('vetorizador.pkl')

def analisar_comentario(texto):
    # Transformando o texto usando o dicionário que o modelo 
    texto_vetorizado = vetorizador.transform([texto])
    
    # Faz a predição
    predicao = model.predict(texto_vetorizado)[0]
    probabilidade = model.predict_proba(texto_vetorizado).max()
    
    sentimento = "Positivo 😊" if predicao == 1 else "Negativo 😡"
    
    print(f"Comentário: {texto}")
    print(f"Sentimento: {sentimento}")
    print(f"Confiança do modelo: {probabilidade:.2%}")
    print("-" * 30)

# --- TESTE PRÁTICO ---
feedback = input("Digite um comentário para testar a IA: ")
analisar_comentario(feedback)


Digite um comentário para testar a IA:  sss


Comentário: sss
Sentimento: Positivo 😊
Confiança do modelo: 55.70%
------------------------------


In [5]:
feedback = input("Digite um comentário para testar a IA: ")
analisar_comentario(feedback)

Digite um comentário para testar a IA:  Que ódio desse produto


Comentário: Que ódio desse produto
Sentimento: Negativo 😡
Confiança do modelo: 67.13%
------------------------------


In [6]:
feedback = input("Digite um comentário para testar a IA: ")
analisar_comentario(feedback)

Digite um comentário para testar a IA:  Odiei esse produto


Comentário: Odiei esse produto
Sentimento: Negativo 😡
Confiança do modelo: 77.15%
------------------------------


# Durante os testes, observou-se que o modelo performa melhor com frases completas do que com palavras isoladas. Por exemplo, a palavra 'ódio' sozinha gera uma confiança baixa (~55%), enquanto 'Odiei esse produto' eleva a confiança para ~77%. Isso valida a escolha técnica dos Bigramas, demonstrando que a IA utiliza a relação entre as palavras para definir o sentimento com maior precisão.